homeassistant/components/knx/notify.py

```diff
@@ -20,24 +21,25 @@ async def async_get_service(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> KNXNotificationService | None:
     """Get the KNX notification service."""
-    if not discovery_info:
```

Comment:
Why do we remove this guard? We still only want to set up the platform via discovery. I'd change it to check for None instead.

Reply:
I readded the guard and check for None instead. We were abusing the discovery info before, that's why I thought its not needed anymore